### ISSUE TITLE
Build assistant add-on to automatically prepare a Firefox profile for Test Pilot development

### ISSUE THREAD

**Issue by lmorchard** [mozilla/testpilot#2840]
It would be nice to have a single legacy add-on to install that does all the cumbersome things needed to have a Firefox profile ready for Test Pilot development.

That includes things such as:

* all of these preferences:
  * extensions.legacy.enabled = true
  * testpilot.env = dev
  * extensions.install.requireBuiltInCerts = false
  * xpinstall.signatures.required = false
  * extensions.webapi.testing = true
* whatever additional prefs that get covered by the legacy [DevPrefs add-on](https://addons.mozilla.org/en-US/firefox/addon/devprefs/) (if we need them)
* optionally installing the [Extension auto-installer](https://addons.mozilla.org/en-US/firefox/addon/autoinstaller/) if that even works anymore after the SDK goes away
  * alternatively, see if the web-ext tool can work for auto-reloading

We have [groundcontrol](https://github.com/mozilla/testpilot/tree/master/groundcontrol) with a toolbar button to help switch between environments. But, that's an SDK add-on and should be considered deprecated. Maybe this would be a good time to rebuild that as an embedded webextension along with adding all the above.

**Comment by lmorchard** [comment]
Ideally the end result of this would be setting extensions.legacy.enabled=true and installing the assistant add-on. Even more ideally would be to get this add-on signed as a Mozilla Extension with the same key as the main Test Pilot add-on and just install it with no prefs tweaks at all

**Comment by lmorchard** [comment]
Not-so-ideally, it probably won't help on release Firefox because the prefs to ignore signatures for Mozilla Extensions will be ignored. So, unbranded builds need to be used to install other non-signed legacy addons, even if the assistant addon *is* signed

**Comment by SoftVision-PaulOiegas** [comment]
A more old fashion way is to create a "user.js" file with the prefs you need and just copy it in the profile folder before starting it. Until a better way is found, we're gonna use that on our end (QA). Five prefs is a lot to remember. 😄

**Comment by chuckharmston** [comment]
The 57 train has sailed, and we can no longer do this.
